sled-agent/src/bootstrap/rss_handle.rs

```diff
@@ -0,0 +1,232 @@
+    let client = bootstrap_agent_client::Client::new_with_client(
+        &url,
+        client,
+        log.new(o!("BootstrapAgentClient" => url.clone())),
+    );
+
+    let sled_agent_initialize = || async {
+        client
+            .start_sled(&bootstrap_agent_client::types::SledAgentRequest {
+                subnet: bootstrap_agent_client::types::Ipv6Subnet {
+                    net: bootstrap_agent_client::types::Ipv6Net(
+                        request.subnet.net().to_string(),
+                    ),
+                },
+            })
+            .await
+            .map_err(BackoffError::transient)?;
+
+        Ok::<
+            (),
+            BackoffError<
+                bootstrap_agent_client::Error<
+                    bootstrap_agent_client::types::Error,
+                >,
+            >,
+        >(())
+    };
+
+    let log_failure = |error, _| {
+        warn!(log, "failed to start sled agent"; "error" => ?error);
+    };
+    retry_notify(internal_service_policy(), sled_agent_initialize, log_failure)
+        .await?;
+    info!(log, "Peer agent at {} initialized", url);
+    Ok(())
+}
+
+// RSS needs to send requests (and receive responses) to and from its local sled
+// agent over some communication channel. Currently RSS lives in-process with
+// sled-agent, so we can use tokio channels. If we move out-of-process we'll
+// need to switch to something like Unix domain sockets. We'll wrap the
+// communication in the types below to avoid using tokio channels directly and
+// leave a breadcrumb for where the work will need to be done to switch the
+// communication mechanism.
+fn rss_sled_agent_channel() -> (RssSledAgentSender, RssSledAgentReceiver) {
+    let (tx, rx) = mpsc::channel(32);
+    (RssSledAgentSender { inner: tx }, RssSledAgentReceiver { inner: rx })
+}
+
+type InnerInitRequest = (
+    Vec<(SocketAddrV6, SledAgentRequest)>,
+    oneshot::Sender<Result<(), String>>,
+);
+
+pub(crate) struct RssSledAgentSender {
+    inner: mpsc::Sender<InnerInitRequest>,
+}
+
+impl RssSledAgentSender {
+    pub(crate) async fn initialize_sleds(
+        &self,
+        requests: Vec<(SocketAddrV6, SledAgentRequest)>,
+    ) -> Result<(), String> {
+        let (tx, rx) = oneshot::channel();
+
+        // IPC will require real error handling, but we know the sled-agent task
+        // will not close the channel until we do and that it will always send a
+        // response, so unwrapping here is fine.
+        self.inner.send((requests, tx)).await.unwrap();
+        rx.await.unwrap()
+    }
+}
+
+struct RssSledAgentReceiver {
+    inner: mpsc::Receiver<InnerInitRequest>,
+}
+
+impl RssSledAgentReceiver {
+    async fn initialize_sleds<F, Fut>(mut self, log: &Logger, init_one_sled: F)
+    where
+        F: Fn(SocketAddrV6, SledAgentRequest) -> Fut,
+        Fut: Future<Output = Result<(), String>>,
+    {
+        let (requests, tx_response) = match self.inner.recv().await {
```

Comment:
Related to my comment above - this looks like it receives "however many requests happened to make it to the buffer", it initializes those sleds, and then the receiver basically stops running